02_snake_animation/main.go

```diff
@@ -48,6 +50,20 @@ func moveSnake(s snake) snake {
 
 // Makes a single iteration for a snake.
 func step(s snake) snake {
+	x, y := s.pos.x, s.pos.y
+	if x == 5 && y == 8 {
+		s.dir.x = 1
+		s.dir.y = 0
+	} else if x == 30 && y == 8 {
+		s.dir.x = 0
+		s.dir.y = 1
+	} else if x == 30 && y == 13 {
+		s.dir.x = -1
+		s.dir.y = 0
+	} else if x == 5 && y == 13 {
+		s.dir.x = 0
+		s.dir.y = -1
+	}
```

Comment:
To avoid many similar ifs, I'd do the following:

```golang
	for _, pivot := range []struct {
		pos, dir coord
	}{
		{coord{5, 8}, coord{0, 1}},
		{coord{30, 8}, coord{0, 1}},
		{coord{30, 13}, coord{-1, 0}},
		{coord{5, 13}, coord{0, -1}},
	} {
		if (coord{x, y} == pivot.pos) {
			s.dir = pivot.dir
		}
	}
```